src/sentry/snuba/metrics.py

```diff
@@ -1190,6 +1193,9 @@ def get_series(self, projects: Sequence[Project], query: QueryDefinition) -> dic
                             Condition(lhs_condition, Op.IN, Function("tuple", condition_value))
                         ]
                     snuba_query = snuba_query.set_where(where)
+                    # Set the limit of the second query to be the provided limits multiplied by
+                    # the number of the metrics requested in the query
+                    snuba_query = snuba_query.set_limit(limit * len(query.fields))
```

Comment:
Just realized that we could even limit it to `limit * number_of_metrics_for_this_entity`, but this is still a good upper bound so let's leave it as it is.

Reply:
You are right!
Its a relatively small change so just corrected it